vendor/bat-native-ledger/src/bat/ledger/internal/database/database_sku_order.cc

```diff
@@ -0,0 +1,312 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/common/bind_util.h"
+#include "bat/ledger/internal/database/database_sku_order.h"
+#include "bat/ledger/internal/database/database_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+
+using std::placeholders::_1;
+
+namespace braveledger_database {
+
+namespace {
+
+const char kTableName[] = "sku_order";
+
+}  // namespace
+
+DatabaseSKUOrder::DatabaseSKUOrder(
+    bat_ledger::LedgerImpl* ledger) :
+    DatabaseTable(ledger),
+    items_(std::make_unique<DatabaseSKUOrderItems>(ledger)) {
+}
+
+DatabaseSKUOrder::~DatabaseSKUOrder() = default;
+
+bool DatabaseSKUOrder::CreateTableV19(ledger::DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  const std::string query = base::StringPrintf(
+      "CREATE TABLE %s ("
+      "order_id TEXT NOT NULL,"
+      "total_amount DOUBLE,"
+      "merchant_id TEXT,"
+      "location TEXT,"
+      "status INTEGER NOT NULL DEFAULT 0,"
+      "contribution_id TEXT,"
+      "created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,"
+      "PRIMARY KEY (order_id)"
+      ")",
+      kTableName);
+
+  auto command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::EXECUTE;
+  command->command = query;
+  transaction->commands.push_back(std::move(command));
+
+  return true;
+}
+
+bool DatabaseSKUOrder::Migrate(
+    ledger::DBTransaction* transaction,
+    const int target) {
+  DCHECK(transaction);
+
+  switch (target) {
+    case 19: {
+      return MigrateToV19(transaction);
+    }
+    default: {
+      return true;
+    }
+  }
+}
+
+bool DatabaseSKUOrder::MigrateToV19(ledger::DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  if (!DropTable(transaction, kTableName)) {
+    return false;
+  }
+
+  if (!CreateTableV19(transaction)) {
```

Comment:
If `DropTable` succeeds and `CreateTableV19` fails, migration will fail. I think create should preceed drop

Reply:
not sure that I understand what you mean

Reply:
I actually mis-interpreted the code. The code is not doing any `data` migration since `sku_order` table is new. My concern was that table is being dropped first and if create fails the data is lost. Sorry for the noise, you can ignore this comment.